ebpf/disk-failure/main.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package main
+
+import (
+	"C"
+	"bytes"
+	"encoding/binary"
+	"flag"
+	"fmt"
+	bpf "github.com/aquasecurity/libbpfgo"
+	"github.com/aquasecurity/libbpfgo/helpers"
+	"os"
+	"os/signal"
+)
+
+var nFlag = flag.Uint64("p", 0, "Process to check")
+var nPath = flag.String("f", "/", "Path to filter")
+
+func main() {
+	flag.Parse()
+
+	sig := make(chan os.Signal, 1)
```

Comment:
I think it would be nice to have some comments in this function to understand everything

Reply:
I have added some new comments and a little bit of refactoring. What do you think of that? @clairecng